### PR TITLE
Adding 'Report abuse' and 'request file' forms

### PR DESCRIPTION
**Pull request by SarahBA**

### CODE REVIEW

File: webui/src/components/reportabuse.jsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
+import React from 'react/lib/ReactWithAddons';
+import { Link, browserHistory } from 'react-router'
+
+import Toggle from 'react-toggle';
+import { DropdownList } from 'react-widgets';
+import { HeightAnimate, ReplaceAnimate } from './animate.jsx';
+
+//TODO: handle the submission
```
Comment (SarahBA) [inline webui/src/components/reportabuse.jsx]: AFAIR it wasn't clear how we should handle it. If there is new information please update me to fix it!

File: webui/src/components/sendform.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react/lib/ReactWithAddons';
```
Comment (emanueldima) [inline webui/src/components/sendform.jsx]: This file seems not to be used at all

File: webui/src/components/reportabuse.jsx (hunk 1 above)
Comment (emanueldima) [inline webui/src/components/reportabuse.jsx]: On submission, the javascript client must send a request to the B2SHARE server. We must add a new HTTP API endpoint (either in /records/ or elsewhere) for this. The server would just pack all the data it gets and send an email to the eudat ticketing system. See for reference this old file: https://github.com/EUDAT-B2SHARE/b2share/blob/1.7/invenio/b2share/modules/abuse/abuse_form.py

File: b2share/modules/records/views.py [hunk 3]
```diff
@@ -273,3 +302,82 @@ def post(self, **kwargs):
...
+                  'affiliation','address','country']:
+            if v not in request.json:
+                msg = v + ' is required'
+                return jsonify({'Error':msg})
+
+        count = 0
+        for ii in ['noresearch', 'abusecontent', 'copyright', 'illegalcontent']:
+            if request.json[ii]: count += 1
+        if count != 1:
+            return jsonify({'Error':'From \'noresearch\', \'abusecontent\', \'copyright\', \'illegalcontent\' (only) one should be True'})
```
Comment (emanueldima) [inline b2share/modules/records/views.py]: must also return an error code

File: b2share/modules/records/views.py [hunk 4]
```diff
@@ -273,3 +302,82 @@ def post(self, **kwargs):
...
+            Phone: """ + str(request.json['phone']) + """
+            """
+        support = str(current_app.config.get('SUPPORT_EMAIL'))
+        msg = Message("Abuse Report for a Record", sender=str(request.json['email']),
+                         recipients=[support], body=msg_content)
+        app = Flask('abuseapp')
+        app.config.update(MAIL_SUPPRESS_SEND=True)
+        InvenioMail(app)
+        with app.app_context():
+            app.extensions['mail'].send(msg)
```
Comment (emanueldima) [inline b2share/modules/records/views.py]: why not `current_app.extensions['mail'].send(msg)`? Why create a new flask app just for that? If it's really needed then it must be documented.

File: b2share/modules/records/views.py [hunk 5]
```diff
@@ -273,3 +302,82 @@ def post(self, **kwargs):
...
+
+
+class RequestAccessResource(RecordResource):
+
+    def post(self, **kwargs):
+        for v in ['message','email','zipcode','phone','city','name',
+                  'affiliation','address','country']:
+            if v not in request.json:
+                msg = v + ' is required'
+                return jsonify({'Error':msg})
```
Comment (emanueldima) [inline b2share/modules/records/views.py]: error code needed

File: b2share/modules/records/views.py [hunk 6]
```diff
@@ -273,3 +302,82 @@ def post(self, **kwargs):
...
+            Phone: """ + str(request.json['phone']) + """
+            """
+        support = str(current_app.config.get('SUPPORT_EMAIL'))
+        msg = Message("Request Access to Data Files", sender=str(request.json['email']),
+                         recipients=[support], body=msg_content)
+        app = Flask('accessapp')
+        app.config.update(MAIL_SUPPRESS_SEND=True)
+        InvenioMail(app)
+        with app.app_context():
+            app.extensions['mail'].send(msg)
```
Comment (emanueldima) [inline b2share/modules/records/views.py]: same question as above, why not use current_app?

File: webui/src/components/record.jsx [hunk 7]
```diff
@@ -215,6 +216,11 @@ const Record = React.createClass({
 
     renderFileList(files) {
         let fileComponent = false;
+        var rec_id = this.props.record.get('id');
+   
+        const user = serverCache.getUser();
+        const request_link = (this.props.record.getIn(['metadata', 'open_access']) === false && this.props.record.get('metadata').get('owners').get(0) != user.get('id')) ? <Link to={`/records/${rec_id}/request`} >Send a request to the owner to get the files</Link> : false;
```
Comment (emanueldima) [inline webui/src/components/record.jsx]: can use `this.props.record.getIn(['metadata', 'owners', 0])` here

File: webui/src/data/server.js [hunk 8]
```diff
@@ -31,6 +31,9 @@ const apiUrls = {
 
     languages()                       { return `${urlRoot}/suggest/languages.json` },
 
+    abusereports(id)                  { return `${urlRoot}/api/records/${id}/abuserecords` },
```
Comment (emanueldima) [inline webui/src/data/server.js]: rename to abuse

File: webui/src/data/server.js [hunk 9]
```diff
@@ -31,6 +31,9 @@ const apiUrls = {
...
+    accessrequestfiles(id)            { return `${urlRoot}/api/records/${id}/accessrequests` },
```
Comment (emanueldima) [inline webui/src/data/server.js]: rename to accessrequests

File: webui/src/components/reportabuse.jsx [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+                            </div>
+                        </div>
+
+                        <div className='row'  style={gap}>
+                            <div className="col-sm-4">
+                                <label forHtml='country'>* Country </label>
+                            </div>
+                            <div className=" col-sm-8" >
+                            <select ref={(ref) => this.country = ref} name='country' className="form-control" defaultValue=""  required >
+                                <option value=""> Choose country </option>
```
Comment (emanueldima) [inline webui/src/components/reportabuse.jsx]: The list of countries should be put in an array and expanded later, it takes too much space and is doubled between /abuse and /accessrequests

**Comment by emanueldima** [comment]
manually rebased and merged, see https://github.com/EUDAT-B2SHARE/b2share/commit/826e7958e4652ceb15ff8468fa1eb535ebc4b771

**Pull request closed without merging**